src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java

```diff
@@ -170,4 +170,28 @@ public Query generateRollupCountsQueryForCriteriaGroupsList(
 
     return new Query(List.of(personId), List.of(rootTable), filterVariable);
   }
+
+  public Query generateRowIdQueryForCriteriaGroupsList(
+      List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList) {
+
+    FieldVariable personId =
+        new FieldVariable(
+            new FieldPointer(getRootTablePointer(), Person.PERSON_ID), rootTable, null, false);
+    FieldVariable rowId =
+        new FieldVariable(new FieldPointer(getRootTablePointer(), Person.ROW_ID), rootTable);
+
+    FilterVariable filterVariable =
+        new BooleanAndOrFilterVariable(
+            BooleanAndOrFilterVariable.LogicalOperator.OR,
+            criteriaGroupsList.stream().map(this::generateFilterForCriteriaGroups).toList());
+
+    FilterVariable subQuery =
+        new SubQueryFilterVariable(
+            personId,
+            SubQueryFilterVariable.Operator.IN,
+            new Query(List.of(personId), List.of(rootTable), filterVariable));
```

Comment:
Could this be refactored to share code with `generateRollupCountsQueryForCriteriaGroupsList()`?